src/storage/flagged.rs

```diff
@@ -60,16 +61,21 @@ use world::{Component, Index};
 ///         self.modified.clear();
 ///         self.inserted.clear();
 ///
-///         // This allows us to use the modification events in a `Join`. Otherwise we
-///         // would have to iterate through the events which may not be in order.
-///         //
-///         // This does not populate the bitset with inserted components, only pre-existing
-///         // components that were changed by a `get_mut` call to the storage.
-///         comps.populate_modified(&mut self.modified_id, &mut self.modified);
-///
-///         // This will only include inserted components from last read, note that this
-///         // will not include `insert` calls if there already was a pre-existing component.
-///         comps.populate_inserted(&mut self.inserted_id, &mut self.inserted);
+///         // Here we can populate the bitsets by iterating over the events.
+///         // You can also just iterate over the events without using a bitset, but this
+///         // allows us to use them in joins with components.
+///         {
+///             let events = comps.channel()
+///                 .read(&mut self.reader_id)
+///                 .collect::<Vec<&ComponentEvent>>();
```

Comment:
Does this really need collect to a `Vec`? Seems odd that it cannot be used directly as iterator.

Reply:
I suppose it could, I was just thinking of the previous time when I had to collect it to use it multiple times.